### DISCUSSION TITLE
Linear layer normalization

### DISCUSSION THREAD

**Discussion by YoavShamir5** [e3nn/e3nn#385] [Q&A]
Hello,

I was confused by the o3.Linear code - what is the default normalization and what are the alternatives? It seems like the default is "element", but I am not sure what is the relevant equations.

Thanks for your help,
Yoav Shamir

**Comment by mariogeiger** [comment]
It's the same parameter as in `TensorProduct`.

```
    path_normalization : {'element', 'path'}
        If set to ``element``, each output is normalized by the total number of elements (independently of their paths).
        If it is set to ``path``, each path is normalized by the total number of elements in the path, then each output is
        normalized by the number of paths.
```

Here is an example for Linear. Imagine you have two lists of vectors of different multiplicities, say `"2x1o + 32x1o"`.

If `path_normalization=element` the linear is implemented as:
```
output = (W1 @ input1 + W2 @ input2) / sqrt(34)
```

If `path_normalization=path` the linear is implemented as:
```
output = (W1 @ input1 / sqrt(2) + W2 @ input2 / sqrt(32)) / sqrt(2)
```
note how in this case each path gets an equal contribution to the output no matter how many inputs contribute to it.

**Reply by YoavShamir5** [reply]
thanks a lot for the info,

Yoav
